core/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -74,32 +73,29 @@ protected ClusterBlockException checkBlock(GetSnapshotsRequest request, ClusterS
     @Override
     protected void masterOperation(final GetSnapshotsRequest request, ClusterState state, final ActionListener<GetSnapshotsResponse> listener) {
         try {
+            final String repository = request.repository();
             List<SnapshotInfo> snapshotInfoBuilder = new ArrayList<>();
             if (isAllSnapshots(request.snapshots())) {
-                snapshotInfoBuilder.addAll(snapshotsService.snapshots(request.repository(), request.ignoreUnavailable()));
+                snapshotInfoBuilder.addAll(snapshotsService.snapshots(repository, request.ignoreUnavailable()));
             } else if (isCurrentSnapshots(request.snapshots())) {
-                snapshotInfoBuilder.addAll(snapshotsService.currentSnapshots(request.repository()));
+                snapshotInfoBuilder.addAll(snapshotsService.currentSnapshots(repository));
             } else {
-                Set<String> snapshotsToGet = new LinkedHashSet<>(); // to keep insertion order
```

Comment:
why did you remove this?
What if we have multiple pattern that match multiple snapshots? we now add these multiple times.
For example:
`foo*,*bar` will match the snapshot `foobar` twice.